src/logging.rs

```diff
@@ -1,31 +1,107 @@
-use chrono::Local;
-use colored::{ColoredString, Colorize};
-use log::SetLoggerError;
+use crate::ProtobufServer::services::query_response::query_ok::Information;
+use log::{LevelFilter, SetLoggerError};
+use simplelog::*;
+use std::fmt::{Arguments, Debug};
 use std::io::Write;
 
+static mut MSGS: Vec<String> = vec![];
+
+// TODO: Implement a logging that works for multiple clients
 #[cfg(feature = "logging")]
-pub fn setup_logger() -> Result<(), SetLoggerError> {
-    fn colored_level(level: log::Level) -> ColoredString {
-        match level {
-            log::Level::Error => level.to_string().red(),
-            log::Level::Warn => level.to_string().yellow(),
-            log::Level::Info => level.to_string().cyan(),
-            log::Level::Debug => level.to_string().blue(),
-            log::Level::Trace => level.to_string().magenta(),
-        }
+/// Sets up the logging
+pub fn setup_logger(is_server: bool) -> Result<(), SetLoggerError> {
+    // For any other areas to be logged, add a `.add_filter_allow_str("clock-reduction")` below here
+    // and specify target when logging - `info!(target: "subject", MSG)`
+    let info_conf = ConfigBuilder::new()
+        .set_time_format_custom(&[])
+        .set_target_level(LevelFilter::Info)
+        .add_filter_allow_str("clock-reduction")
+        .build();
+    let loggers: Vec<Box<dyn SharedLogger>> = vec![
+        TermLogger::new(
+            LevelFilter::Warn,
+            Config::default(),
+            TerminalMode::Mixed,
+            ColorChoice::Auto,
+        ),
+        if is_server {
+            WriteLogger::new(LevelFilter::Info, info_conf, G {})
+        } else {
+            TermLogger::new(
+                LevelFilter::Info,
+                info_conf,
+                TerminalMode::Mixed,
+                ColorChoice::Auto,
+            )
+        },
+    ];
+
+    CombinedLogger::init(loggers)
+}
+
+fn get_messages_raw() -> impl Iterator<Item = String> + Sized {
+    unsafe {
```

Comment:
Can this be something other than unsafe?

Reply:
no, when dealing with `static mut` it must be in an `unsafe` function or block

Reply:
Is it possible to use something other than `static mut`?

Reply:
Not really